flake.nix

```diff
@@ -92,14 +92,24 @@
     in eachDefaultSystem (system:
       let pkgs = import nixpkgs { inherit system; };
       in {
+        apps = {
+          default = self.outputs.apps.${system}.nix-doom-emacs;
+          nix-doom-emacs = flake-utils.lib.mkApp {
+            drv = self.outputs.packages.${system}.nix-doom-emacs;
+            exePath = "/bin/emacs";
+          };
+        };
+
```

Comment:
Perhaps hold off on this until we can make the "get started with doom!"-esque thing from https://github.com/nix-community/nix-doom-emacs/pull/266#discussion_r967800252?

Reply:
I think it is better to implement it here and then document it in #266.

Reply:
"it" is this PR.

Now, my idea of having `apps` output here is only to allow users to have a taste of having a fully functional doom-emacs without users even needing to setup themselves. We also basically get this for free since we already have the `package.default` output, and the only thing we need here is setting the correct `exePath`. So I think it is worth having.

Reply:
But the [default config](https://github.com/nix-community/nix-doom-emacs/blob/da0ce88c0a0a3950a3b215e7a9f1573c555413e5/test/doom.d/init.el#L86) has a lot of things disabled. It's not fully functional if they can't just `SPC p p` their project and use it – or see how easy it [usually] is to uncomment a line and rebuild. 

Perhaps we should build a doom with All The Things enabled and set up? `doom-full`?

(Weird how we call the app and package `nix-doom-emacs` when in fact they're `doom-emacs`)

Reply:
It seems like a lot of work to make a functional `doom-full`. Like look, here's my [init.el](https://github.com/ckiee/nixfiles/blob/a046c6969867133b01d9b62a6db27acbaab523a0/modules/doom-emacs/config/init.el) and here's all the things that need to be in [PATH](https://github.com/ckiee/nixfiles/blob/a046c6969867133b01d9b62a6db27acbaab523a0/modules/doom-emacs/default.nix#L6-L44) for my work.

Reply:
Trying to turn NDE into *An Experience* seems like a lot more work than I'm willing to do; I'm fine with it being a lego brick you slot into your configuration and build with. Both my [Elisp config](https://github.com/ckiee/nixfiles/blob/a046c6969867133b01d9b62a6db27acbaab523a0/modules/doom-emacs/config/config.org) and [the nix code](https://github.com/ckiee/nixfiles/blob/a046c6969867133b01d9b62a6db27acbaab523a0/modules/doom-emacs/default.nix) have a lot of quirks suited for specifically me.

(:thread: 3/3)

Reply:
> But the [default config](https://github.com/nix-community/nix-doom-emacs/blob/da0ce88c0a0a3950a3b215e7a9f1573c555413e5/test/doom.d/init.el#L86) has a lot of things disabled. It's not fully functional if they can't just SPC p p their project and use it – or see how easy it [usually] is to uncomment a line and rebuild.

Yeah, I know. And this is by purpose. This is meant as an example on what you can get. It is not my idea here to have a full configuration working for all kinda of usages.

The idea of having `nix run` working is just for the user see "ohh, so I can a fully function doom-emacs without needing to enabling anything else", and them they will add it to their configuration and customize on their own.

Reply:
Perhaps we should customize doom-dashboard to explain that this is like a demo? And enable `nix` for it.. :P

Reply:
(Actually, the default config also supports [sh](https://github.com/nix-community/nix-doom-emacs/blob/master/test/doom.d/init.el#L167) and [markdown](https://github.com/nix-community/nix-doom-emacs/blob/master/test/doom.d/init.el#L149)).

> Perhaps we should customize doom-dashboard to explain that this is like a demo? And enable `nix` for it.. :P

The problem of enabling anything in the config is that it will be overwrite by the update script. I did this in the past :P.

Also, doom-dashboard is not working for ages, unless you use `emacsGit` (that is also a bad idea).

Reply:
It works for me with emacs as a daemon but I'm behind on the NDE source.

Reply:
> Now, my idea of having apps output here is only to allow users to have a taste of having a fully functional doom-emacs without users even needing to setup themselves.

This is an awesome feature to have. I can show Doom to my work colleagues this way!